cylc/flow/suite_srv_files_mgr.py

```diff
@@ -33,622 +34,672 @@
-    KEY_HOST = "CYLC_SUITE_HOST"
-    KEY_NAME = "CYLC_SUITE_NAME"
-    KEY_OWNER = "CYLC_SUITE_OWNER"
-    KEY_PROCESS = "CYLC_SUITE_PROCESS"
-    KEY_PORT = "CYLC_SUITE_PORT"
-    KEY_SSH_USE_LOGIN_SHELL = "CYLC_SSH_USE_LOGIN_SHELL"
-    KEY_SUITE_RUN_DIR_ON_SUITE_HOST = "CYLC_SUITE_RUN_DIR_ON_SUITE_HOST"
-    KEY_TASK_MSG_MAX_TRIES = "CYLC_TASK_MSG_MAX_TRIES"
-    KEY_TASK_MSG_RETRY_INTVL = "CYLC_TASK_MSG_RETRY_INTVL"
-    KEY_TASK_MSG_TIMEOUT = "CYLC_TASK_MSG_TIMEOUT"
-    KEY_UUID = "CYLC_SUITE_UUID"
-    KEY_VERSION = "CYLC_VERSION"
-    NO_TITLE = "No title provided"
-    PASSPHRASE_CHARSET = ascii_letters + digits
-    PASSPHRASE_LEN = 20
-    PS_OPTS = '-opid,args'
-    REC_TITLE = re.compile(r"^\s*title\s*=\s*(.*)\s*$")
-
-    def __init__(self):
-        self.local_passphrases = set()
-        self.cache = {self.FILE_BASE_PASSPHRASE: {}}
-        self.can_disk_cache_passphrases = {}
-        self.can_use_load_auths = {}
-
-    def cache_passphrase(self, reg, owner, host, value):
-        """Cache and dump passphrase for a remote suite in standard location.
-
-        Save passphrase to ~/.cylc/auth/owner@host/reg if possible.
-        This is normally called on a successful authentication, and will cache
-        the remote passphrase in memory as well.
-        """
-        if owner is None:
-            owner = get_user()
-        if host is None:
-            host = get_host()
-        path = self._get_cache_dir(reg, owner, host)
-        self.cache[self.FILE_BASE_PASSPHRASE][(reg, owner, host)] = value
-        # Dump to a file only for remote suites loaded via SSH.
-        if self.can_disk_cache_passphrases.get((reg, owner, host)):
-            # Although not desirable, failing to dump the passphrase to a file
-            # is not disastrous.
-            try:
-                self._dump_item(path, self.FILE_BASE_PASSPHRASE, value)
-            except (IOError, OSError):
-                if cylc.flow.flags.debug:
-                    import traceback
-                    traceback.print_exc()
-
-    def detect_old_contact_file(self, reg, check_host_port=None):
-        """Detect old suite contact file.
-
-        If an old contact file does not exist, do nothing. If one does exist
-        but the suite process is definitely not alive, remove it. If one exists
-        and the suite process is still alive, raise SuiteServiceFileError.
-
-        If check_host_port is specified and does not match the (host, port)
-        value in the old contact file, raise AssertionError.
-
-        Args:
-            reg (str): suite name
-            check_host_port (tuple): (host, port) to check against
-
-        Raise:
-            AssertionError:
-                If old contact file exists but does not have matching
-                (host, port) with value of check_host_port.
-            SuiteServiceFileError:
-                If old contact file exists and the suite process still alive.
+class SuiteFiles:
```

Comment:
For future work this class should hold the full Cylc tree (e.g. `log`, `work`, `job`, etc).

We could then use this class to auto-generate the documentation.

Reply:
(Post an issue so we don't forget this?)

Reply:
Done #3345